File: Leetcode/array/PlusOne/src/python/PlusOne.py
# python2 代码
class Solution:
    def plusOne(self,digits):
        dig=1;
        for i in reversed(xrange(len(digits))):
            digits[i]+=dig
            dig=digits[i]/10;
            digits[i]%=10
        if dig:
            digits=[1]+digits
        return digits
if __name__=="__mian__":
    print Solution().plusOne([9,9,9,9])
    
# python3 代码
class Solution:
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        return self.plusX(digits, 1)
    def plusX(self, digits, dig):
        for i in reversed(range(len(digits))):
            sum=digits[i]+dig
            dig=sum//10
            digits[i]=sum%10
        if dig:
            digits=[1]+digits
        return digits

File: Leetcode/array/PlusOne/src/cpp/PlusOne.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        plus(digits,1);
        return digits;
    }
private:
    void plus(vector<int>& digits,int dig){
        int sum=0;
        for(int i=digits.size()-1;i>=0;i--){
            sum=digits[i]+dig;
            digits[i]=sum%10;
            dig=sum/10;
        }
        if (dig)
            digits.insert(digits.begin(),1);
    }
};